dhall/src/Dhall/Eval.hs

```diff
@@ -603,33 +605,38 @@ eval !env t0 =
                 t                       -> VTextShow t
         TextReplace ->
             VPrim $ \needle ->
-            VPrim $ \replacement ->
-            VPrim $ \haystack ->
-                case needle of
-                    VTextLit (VChunks [] "") ->
-                        haystack
-                    VTextLit (VChunks [] needleText) ->
-                        case haystack of
-                            VTextLit (VChunks [] haystackText) ->
-                                case replacement of
-                                    VTextLit (VChunks [] replacementText) ->
-                                        VTextLit $ VChunks []
-                                            (Text.replace
-                                                needleText
-                                                replacementText
-                                                haystackText
-                                            )
+                let hLamInfo = case needle of
+                        VTextLit (VChunks [] "") -> TextReplaceEmpty
+                        _                        -> Prim
+
+                in  VHLam hLamInfo $ \replacement ->
```

Comment:
So that's how you include the lambda in the `TextReplaceEmpty` case! :) I hadn't figured that out.